pykokkos/core/type_inference/args_type_inference.py

```diff
@@ -50,6 +52,8 @@ class UpdatedDecorator:
 # DataType class has all supported pk datatypes, we ignore class members starting with __, add enum duplicate aliases
 SUPPORTED_NP_DTYPES = [attr for attr in dir(DataType) if not attr.startswith("__")] + ["float64", "float32"]
 
+# Cache for original argument nodes
+ORIGINAL_PARAMS: Dict[str, ast.arguments] = {}
```

Comment:
Leave a more detailed comment on what this is mapping from. What exactly does the `str` represent here?